src/Products/PageTemplates/tests/testExpressions.py

```diff
@@ -90,6 +89,15 @@ def test_evaluate_with_render_simple_callable(self):
         ec = self._makeContext()
         self.assertEqual(ec.evaluate('dummy'), 'dummy')
 
+    def test_evaluate_with_unimplemented_call(self):
+        class Dummy(object):
+            def __call__(self):
+                raise NotImplementedError
```

Comment:
I'd suggest to raise an instance of `NotImplementedError` instead of the class so Python 3 can store the traceback on it without side effects.

Reply:
Interestingly, [Python automatically turns](https://github.com/python/cpython/blob/master/Python/ceval.c#L4140) `raise <type>` into `raise <type>()`---that is, if raise is given a class argument, Python creates a fresh instance and raises that. This turns out to be just a tiny bit faster than constructing the instance yourself (it's one bytecode shorter).

Reply:
@jamadden Thank you for clarifying this.